.github/workflows/tests.yml

```diff
@@ -0,0 +1,42 @@
+name: Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```

Comment:
should we run it on all PR instead?

Reply:
Can do, that's not a problem, it's just usually 'convention' to only run on PR's to the main branch (at least from what I've seen).

Reply:
Let's run on all PRs. It doesn't come up often, but if I make a secondary PR against a branch other than master, it's still useful to have CI run on it.

(I also tend to have it run on push for all branches, but I appreciate that's kind of wasteful if you're just going to make a PR and run the same tests anyway)

Reply:
Removed the restriction, now it runs on a push to master, and any PR